listings/applications/components/src/ownable.cairo

```diff
@@ -93,3 +96,146 @@ pub mod ownable_component {
         }
     }
 }
+// ANCHOR_END: component
+
+#[starknet::contract]
+mod MockContract {
+    use super::{IOwnable, ownable_component, ownable_component::OwnableInternalTrait};
+
+    component!(path: ownable_component, storage: ownable, event: OwnableEvent);
+
+    #[abi(embed_v0)]
+    impl OwnableImpl = ownable_component::Ownable<ContractState>;
+
+    #[storage]
+    struct Storage {
+        #[substorage(v0)]
+        ownable: ownable_component::Storage,
+    }
+
+    #[constructor]
+    fn constructor(ref self: ContractState) {
+        self.ownable._init(starknet::get_caller_address());
+    }
+
+    #[event]
+    #[derive(Drop, starknet::Event)]
+    enum Event {
+        #[flat]
```

Comment:
```suggestion
```
I know I was the one who suggested you include this attribute, but I actually found out how to make the tests pass *without* it (see https://github.com/NethermindEth/StarknetByExample/pull/205/files#r1657310788).

The issue with `#[flat]` is that you don't _always want to flatten your event structure, as it might cause name conflicts_. So we need a way to test the event emissions without flattening them.